### PR TITLE
Improve backup and restore procedures in xtrabackup plugin

### PR DESCRIPTION
**Pull request by plafosse-orange**
At Orange FR, we backup our MySQL instances via the xtrabackup plugin. It's working properly. But when the backup or the restore fail, we have no trace of the error in the logfile. 

In this PR work, we suggest to make some preliminary tests and to indicate in the log file the result of these tests.
For the backup, we test the existence of the backup temporary directory.

For the restore, we test 
- that the target Mysql instance is stopped
- the existence of the backup temporary directory
- the existence of the MySQL data directory
- and that the MySQL data directory is empty

Also, at the end of restore, we assign the uid-gid couple of the MySQL data directory to the contents of this directory

Don't hesitate to suggest any improvements to this work!

Cheers

### CODE REVIEW

File: plugin/xtrabackup/plugin.go [hunk 1]
```diff
@@ -50,7 +50,7 @@
 // mysql_temp_targetdir:
 // This option specifies the absolute path to a temporary directory used by
 // the `xtrabackup` tool to backup the MySQL databases. It must be empty after
-// each run of the plugin.
+// each run of the plugin.i It must be as big as the estimated MySQL data directory.
```
Comment (jhunt) [inline plugin/xtrabackup/plugin.go]: You have an extra 'i' here (vim?)

File: plugin/xtrabackup/plugin.go [hunk 2]
```diff
@@ -76,8 +79,9 @@ package main
 
 import (
 	"os"
-
 	fmt "github.com/jhunt/go-ansi"
+	"path/filepath"
+        "syscall"
```
Comment (jhunt) [inline plugin/xtrabackup/plugin.go]: You should `go fmt` this file.

Also, we group our imports into three sections:

```
import (
  /* standard library imports */

  /* external libraries */

  /* modules internal to SHIELD (but still on Github) */
)
```

Each group should be separated by blank lines (two `\n`'s)

File: plugin/xtrabackup/plugin.go [hunk 3]
```diff
@@ -302,15 +320,18 @@ func (p XtraBackupPlugin) Backup(endpoint plugin.ShieldEndpoint) error {
 
 	plugin.DEBUG("Executing: `%s`", cmdString)
 	if err = plugin.ExecWithOptions(opts); err != nil {
+		fmt.Fprintf(os.Stderr,"@R{\u2717 Create backup files}\n")
 		return err
 	}
+	fmt.Fprintf(os.Stderr,"@G{\u2713 Create backup files}\n")
```
Comment (jhunt) [inline plugin/xtrabackup/plugin.go]: These log message read better as "Creating backup files failed: <err>" and "Created backup files"

File: plugin/xtrabackup/plugin.go [hunk 4]
```diff
@@ -302,15 +320,18 @@ func (p XtraBackupPlugin) Backup(endpoint plugin.ShieldEndpoint) error {
...
 
 	// create and return archive
 	cmdString = fmt.Sprintf("%s -cf - -C %s .", xtrabackup.Tar, targetDir)
 	if err = plugin.Exec(cmdString, plugin.STDOUT); err != nil {
+		fmt.Fprintf(os.Stderr,"@R{\u2717 Create archive}\n")
 		return err
 	}
-
+	fmt.Fprintf(os.Stderr,"@G{\u2713 Create archive}\n")
```
Comment (jhunt) [inline plugin/xtrabackup/plugin.go]: These log messages read better as "Creating archive failed: <err>" and "Created archive"

**Comment by plafosse-orange** [comment]
The Xtrabackup plugin is modified with your comments.

**Comment by jhunt** [comment]
LGTM.  Thanks!

**Pull request merged**